classes/class-wc-connect-service-settings-store.php

```diff
@@ -124,15 +124,114 @@ public function set_preferred_paper_size( $size ) {
 			return WC_Connect_Options::update_option( 'paper_size', $size );
 		}
 
+		/**
+		 * Attempts to recover faulty json string fields that might contain strings with unescaped quotes
+		 *
+		 * @param string $field_name
+		 * @param string $json
+		 *
+		 * @return string
+		 */
+		public function try_recover_invalid_json_string( $field_name, $json ) {
+			$regex = '/"' . $field_name . '":"(.+?)","/';
+			preg_match_all( $regex, $json, $match_groups );
+			if ( 2 === count( $match_groups ) ) {
+				foreach ( $match_groups[ 0 ] as $idx => $match ) {
+					$value = $match_groups[ 1 ][ $idx ];
+					$escaped_value = preg_replace( '/(?<!\\\)"/', '\\"', $value );
+					$json = str_replace( $match, '"' . $field_name . '":"' . $escaped_value . '","', $json );
+				}
+			}
+			return $json;
+		}
+
+		/**
+		 * Attempts to recover faulty json string array fields that might contain strings with unescaped quotes
+		 *
+		 * @param string $field_name
+		 * @param string $json
+		 *
+		 * @return string
+		 */
+		public function try_recover_invalid_json_array( $field_name, $json ) {
+			$regex = '/"' . $field_name . '":\["(.+?)"\]/';
+			preg_match_all( $regex, $json, $match_groups );
+			if ( 2 === count( $match_groups ) ) {
+				foreach ( $match_groups[ 0 ] as $idx => $match ) {
+					$array = $match_groups[ 1 ][ $idx ];
+					$escaped_array = preg_replace( '/(?<![,\\\])"(?!,)/', '\\"', $array );
+					$json = str_replace( $array, $escaped_array, $json );
+				}
+			}
+			return $json;
+		}
+
+		/**
+		 * Returns labels for the specific order ID
+		 *
+		 * @param $order_id
+		 *
+		 * @return array
+		 */
+		public function get_label_order_meta_data( $order_id ) {
+			$label_data = get_post_meta( ( int ) $order_id, 'wc_connect_labels', true );
+			//return an empty array if the data doesn't exist
+			if ( ! $label_data ) {
+				return array();
+			}
+
+			//labels stored as an array, return
+			if ( ! is_string( $label_data ) ) {
```

Comment:
The comment claims to test for an array, but the conditional just checks "not string". Why not `is_array()` here?